code/modules/mob/mob_movement.dm

```diff
@@ -276,15 +276,17 @@
 				move_delay += 7+config.walk_speed
 		move_delay += mob.movement_delay()
 
+		if(locate(/obj/item/weapon/grab, mob))

```

Comment:
Locate is pretty expensive, and there's no point in calling it twice when it could be called once.